index.js

```diff
@@ -147,43 +166,54 @@ module.exports = async function (url, options) {
 
     return new Promise(function (resolve, reject) {
         const wrappedFetch = async () => {
-            ++attempt;
-
-            let timeoutHandler;
-            if (retryOptions.socketTimeout) {
-                const controller = new AbortController();
-                timeoutHandler = setTimeout(() => controller.abort(), retryOptions.socketTimeout);
-                options.signal = controller.signal;
-            }
-
-            try {
-                const response = await fetch(url, options);
-                clearTimeout(timeoutHandler);
-
-                if (!retry(retryOptions, null, response)) {
-                    // response.timeout should reflect the actual timeout
-                    response.timeout = retryOptions.socketTimeout;
-                    return resolve(response);
-                }
-
-                console.error(`Retrying in ${retryOptions.retryInitialDelay} milliseconds, attempt ${attempt - 1} failed (status ${response.status}): ${response.statusText}`);
-            } catch (error) {
-                clearTimeout(timeoutHandler);
-
-                if (!retry(retryOptions, error, null)) {
-                    if (error.name === 'AbortError') {
-                        return reject(new FetchError(`network timeout at ${url}`, 'request-timeout'));
+            let response;
+            while (!isResponseTimedOut(retryOptions)) {
+                ++attempt;
+
+                let timeoutHandler;
+                if (retryOptions.socketTimeout) {
```

Comment:
Is `retryOptions` always set? If not (null/undefined), it would crash here.

Reply:
+1, looks like retryOptions (set by retryInit:https://github.com/adobe/node-fetch-retry/blob/8635a49989e3afa8227f3ed5c855e7059f2df6e6/index.js#L163) could return `false`: https://github.com/adobe/node-fetch-retry/blob/8635a49989e3afa8227f3ed5c855e7059f2df6e6/index.js#L97